File: src/main/java/org/seasar/doma/it/holder/Salary.java
package org.seasar.doma.it.holder;

import java.math.BigDecimal;

import org.seasar.doma.Holder;

@Holder(valueType = BigDecimal.class)
public class Salary {

    private final BigDecimal value;

    public Salary(BigDecimal value) {
        this.value = value;
    }

    public Salary(String value) {
        this(new BigDecimal(value));
    }

    public BigDecimal getValue() {
        return value;
    }
}

File: kotlin/src/main/kotlin/org/seasar/doma/it/kotlin/criteria/Salary.kt
package org.seasar.doma.it.kotlin.criteria

import org.seasar.doma.Domain
import java.math.BigDecimal

@Domain(valueType = BigDecimal::class)
class Salary(val value: BigDecimal) {

    constructor(value: String?) : this(BigDecimal(value)) {}

    override fun toString(): String {
        return value.toString()
    }
}
